src/compas_xr/project/project_manager.py

```diff
@@ -0,0 +1,418 @@
+import os
+
+from compas.geometry import Frame
+from compas_timber.assembly import TimberAssembly
+from compas_timber.planning import BuildingPlan
+from compas_timber.planning import Step
+
+from compas_xr.project.assembly_extensions import AssemblyExtensions
+from compas_xr.realtime_database import RealtimeDatabase
+from compas_xr.storage import Storage
+
+
+class ProjectManager(object):
+    """
+    The ProjectManager class is responsible for managing project specific data and operations that involve
+    Firebase Storage and Realtime Database configuration.
+
+    Parameters
+    ----------
+    config_path : str
+        The path to the configuration file for the project.
+
+    Attributes
+    ----------
+    storage : Storage
+        The storage instance for the project.
+    database : RealtimeDatabase
+        The realtime database instance for the project.
+    """
+
+    def __init__(self, config_path):
+        if not os.path.exists(config_path):
+            raise Exception("Could not create Storage or Database with path {}!".format(config_path))
+        self.storage = Storage(config_path)
+        self.database = RealtimeDatabase(config_path)
+
+    def application_settings_writer(self, project_name, storage_folder="None", z_to_y_remap=False):
+        """
+        Uploads required application settings to the Firebase RealtimeDatabase.
+
+        Parameters
+        ----------
+        project_name : str
+            The name of the project where the app will look for information.
+        storage_folder : str, optional
+            The name of the storage folder, by default "None"
+        z_to_y_remap : bool, optional
+            The orientation of the object, if the obj was exported with z to y remap, by default False
+
+        Returns
+        -------
+        None
+
+        """
+        data = {"project_name": project_name, "storage_folder": storage_folder, "z_to_y_remap": z_to_y_remap}
+        self.database.upload_data(data, "ApplicationSettings")
+
+    def create_project_data_from_compas(self, assembly, building_plan, qr_frames_list):
+        """
+        Formats data structure from Compas Class Objects.
+
+        Parameters
+        ----------
+        assembly : compas.datastructures.Assembly or compas_timber.assembly.TimberAssembly
+            The assembly in which data will be extracted from.
+        building_plan : compas_timber.planning.BuildingPlan
```

Comment:
If you want these things to be clickable links to the doc API, they need to be written like this:

```suggestion
        building_plan : :class:`compas_timber.planning.BuildingPlan`
```